Solutions/CohesitySecurity/Data Connectors/Helios2Sentinel/IncidentProducer/IncidentProducer.cs

```diff
@@ -0,0 +1,252 @@
+            long sev = 70;
+
+            foreach (var prop in alert.propertyList)
+            {
+                switch (i)
+                {
+                case 0:
+                case 4:
+                case 7:
+                case 10:
+                case 11:
+                    TextWriter(id, (string)prop.key, (string)prop.value, log);
+                    break;
+                case 1:
+                    output.properties.title += ". Object: " + prop.value;
+                    TextWriter(id, (string)prop.key, (string)prop.value, log);
+                    break;
+                case 3:
+                    output.properties.title += ". Source: " + prop.value;
+                    break;
+                case 12:
+                    sev = long.Parse((string)prop.value);
+
+                    if (sev >= 70)
+                        output.properties.severity = "High";
+                    else if (sev < 30)
+                        output.properties.severity = "Low";
+                    else
+                        output.properties.severity = "Medium";
+                    break;
+                }
+                i++;
+                if (i == 13) break;
+            }
+            output.properties.Description = alert.alertDocument.alertDescription + ". Alert cause: " + alert.alertDocument.alertCause + ". Anomaly Strength: " + sev + ". Additional Info: " + alert.alertDocument.alertHelpText + ". Helios ID: " + alert.id;
+
+            lock (queueLock)
+            {
+                outputQueueItem.Add(JsonConvert.SerializeObject(output));
+            }
+        }
+
+        private static string GetData(string path)
+        {
+            if (CloudStorageAccount.TryParse(blobStorageConnectionString, out var storageAccount))
+            {
+                // Create the CloudBlobClient that represents the Blob storage endpoint for the storage account.
+                var cloudBlobClient = storageAccount.CreateCloudBlobClient();
+                var container = cloudBlobClient.GetContainerReference(containerName);
+                var blobRef = container.GetBlockBlobReference(path);
+
+                using (var ms = new MemoryStream())
+                    using (var sr = new StreamReader(ms))
+                    {
+                        var task = blobRef.DownloadToStreamAsync(ms);
+                        task.Wait();
+                        ms.Position = 0;
+                        return sr.ReadToEnd();
+                    }
+            }
+            else
+            {
+                throw new InvalidProgramException("Invalid format string");
+            }
+        }
+
+        private static void WriteData(string path, string data, ILogger log)
+        {
+            try
+            {
+                if (CloudStorageAccount.TryParse(blobStorageConnectionString, out var storageAccount))
+                {
+                    var cloudBlobClient = storageAccount.CreateCloudBlobClient();
```

Comment:
The values for both of these variables never change after initialization. It's worth putting them outside of the function
```
                var cloudBlobClient = storageAccount.CreateCloudBlobClient();
                var container = cloudBlobClient.GetContainerReference(containerName);
```